src/isd/batch.py

```diff
@@ -81,6 +82,8 @@ def to_json(self, indent: int = 4) -> str:
             data.append(d)
         return json.dumps(data, indent=indent)
 
-    def to_data_frame(self) -> pandas.DataFrame:
+    def to_data_frame(self) -> "pandas.DataFrame":
         """Reads a local ISD file into a DataFrame."""
+        import pandas
```

Comment:
```suggestion
        try:
            import pandas
        except ImportError as err:
            message = "The `pandas` optional dependency is required to use `to_data_frame`. Install this dependency with `pip install 'isd[pandas]'`.
            raise ImportError("{}\n{}", message, err)
```

That might not be exactly correct, but you get the gist.